scripts/create_test_requirements.sh

```diff
@@ -3,8 +3,9 @@ if [ -z "$(ls | grep CONTRIBUTING.md)" ]; then
   echo "Please run the script from repo directory"
   exit -1
 else
-  echo "Creating requirements under test/requirements.txt using requirements.in. Please run this script from unix or wsl!"
+  echo "Creating requirements under test/requirements.txt using requirements.in. Please run this script from unix or wsl in a python3.7 env!"
+  python -c "import sys; assert sys.version_info < (3, 8) and sys.version_info >= (3, 7), 'Run from python 3.7!'" || exit 1
```

Comment:
Modified this script to exit if it's not running in a python 3.7 environment so that devs do not create a requirements file that cannot be installed in python 3.7.

![image](https://user-images.githubusercontent.com/41651716/179620526-5991c180-7515-4401-a348-8ceb9b80dc3a.png)

Reply:
Hmm am I correct in in understanding that this *requires* our devs to be running 3.7 when they run this script? I don't think we should require that since the vast, vast majority of developers will be running a newer version of Python and this would require them to install an older version. 

Instead, the approach that we should do is to simply check whether the test-requirements can be installed in a 3.7 environment. We should set up a CI environment that is running Python 3.7 and just confirm that the requirements and test-requirements can be installed there.

Reply:
Spoke with @abidlabs about this off line and instead we'll let this script run in any python version.

If devs push deps versions that are not compatible in python 3.7 they will know because the ci will fail.